pywinauto/uia_element_info.py

```diff
@@ -259,15 +259,34 @@ def children(self, **kwargs):
         cond = IUIA().build_condition(**kwargs)
         return self._get_elements(IUIA().tree_scope["children"], cond, cache_enable)
 
+    def has_depth(self, root, depth):
+        if self != root:
+            if depth > 0:
+                parent = self.parent
+                return parent.has_depth(root, depth - 1)
+            else:
+                return False
+        else:
+            return True
+
     def descendants(self, **kwargs):
         """Return a list of all descendant children of the element
 
          * **kwargs** is a criteria to reduce a list by process,
            class_name, control_type, content_only and/or title.
         """
         cache_enable = kwargs.pop('cache_enable', False)
+        depth = kwargs.pop('depth', None)
         cond = IUIA().build_condition(**kwargs)
-        return self._get_elements(IUIA().tree_scope["descendants"], cond, cache_enable)
+        elements = self._get_elements(IUIA().tree_scope["descendants"], cond, cache_enable)
+
+        if depth is not None:
+                if isinstance(depth, int) and depth > 0:
+                    elements = [element for element in elements if element.has_depth(self, depth)]
+                else:
+                    raise Exception("Depth must be natural number")
+
```

Comment:
Same here, DRY. The whole "if" statement can be moved into a separate protected function